python/ribasim/tests/test_model.py

```diff
@@ -362,3 +364,79 @@ def test_model_diff(basic):
     assert "static" in x["basin"]
     assert "diff" in x["basin"]["static"]
     assert isinstance(x["basin"]["static"]["diff"], datacompy.Compare)
+
+
+def test_version_mismatch_warning_newer_version(basic, tmp_path):
+    """Test that a warning is issued when TOML ribasim_version is newer than package version."""
+    # Write the model to get a TOML file
+    toml_path = tmp_path / "ribasim.toml"
+    basic.write(toml_path)
+
+    # Read the TOML file and modify the version to be newer
+    with open(toml_path, "rb") as f:
+        config = tomli.load(f)
+
+    # Set a newer version
+    config["ribasim_version"] = "3030.1.0"
+
+    # Write the modified TOML back
+    with open(toml_path, "wb") as f:
+        tomli_w.dump(config, f)
+
+    # Test that a warning is issued when reading the model
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
You could also use https://docs.pytest.org/en/stable/how-to/capture-warnings.html#warns

Reply:
Ah yes much shorter.